src/operator/src/statement/ddl.rs

```diff
@@ -518,8 +518,54 @@ impl StatementExecutor {
             expr
         );
 
-        let req = SubmitDdlTaskRequest {
-            task: DdlTask::new_alter_table(expr.clone()),
+        let physical_table_id = self
+            .table_metadata_manager
+            .table_route_manager()
+            .get_physical_table_id(table_id)
+            .await
+            .context(TableMetadataManagerSnafu)?;
+
+        let (req, invalidate_keys) = if physical_table_id == table_id {
+            // This is physical table
+            let req = SubmitDdlTaskRequest {
+                task: DdlTask::new_alter_table(expr),
+            };
+
+            let mut invalidate_keys = vec![
+                CacheIdent::TableId(physical_table_id),
+                CacheIdent::TableId(table_id),
+                CacheIdent::TableName(TableName::new(catalog_name, schema_name, table_name)),
+            ];
+
+            let physical_table = self
+                .table_metadata_manager
+                .table_info_manager()
+                .get(physical_table_id)
+                .await
+                .context(TableMetadataManagerSnafu)?
+                .map(|x| x.into_inner());
+            if let Some(physical_table) = physical_table {
+                let physical_table_name = TableName::new(
+                    physical_table.table_info.catalog_name,
+                    physical_table.table_info.schema_name,
+                    physical_table.table_info.name,
+                );
+                invalidate_keys.push(CacheIdent::TableName(physical_table_name));
+            }
```

Comment:
question: When does `physical_table.table_info` have different info than `table.table_info()` we get from the catalog manager above?

Reply:
The `table.table_info` belongs to a logical table. When we alter a logical table, it may also update its corresponding physical table. Therefore, we need to invalidate the cache for the physical table.

Reply:
But we checked `physical_table_id == table_id` before.